crates/rethnet_eth/src/receipt/transaction.rs

```diff
@@ -0,0 +1,113 @@
+use std::ops::Deref;
+
+use ethbloom::Bloom;
+use revm_primitives::{Address, B256, U256};
+
+use super::TypedReceipt;
+
+/// Type for a receipt that's created when processing a transaction.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
+pub struct TransactionReceipt<L> {
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    pub inner: TypedReceipt<L>,
+    /// Hash of the transaction
+    pub transaction_hash: B256,
+    /// Index of the transaction in the block
+    #[serde(with = "crate::serde::u64")]
+    pub transaction_index: u64,
+    /// Address of the sender
+    pub from: Address,
+    /// Address of the receiver. null when it's a contract creation transaction.
+    pub to: Option<Address>,
+    /// The contract address created, if the transaction was a contract creation, otherwise null.
+    pub contract_address: Option<Address>,
+    /// Gas used by this transaction alone.
+    #[serde(serialize_with = "crate::serde::u256::serialize")]
+    pub gas_used: U256,
+    /// The actual value per gas deducted from the senders account. Before EIP-1559, this is equal
+    /// to the transaction's gas price. After, it is equal to baseFeePerGas + min(maxFeePerGas -
+    /// baseFeePerGas, maxPriorityFeePerGas).
+    #[serde(serialize_with = "crate::serde::u256::serialize")]
+    pub effective_gas_price: U256,
+}
+
+impl<L> TransactionReceipt<L> {
+    /// Returns the gas used by the transactions up until this point.
+    pub fn cumulative_gas_used(&self) -> U256 {
+        self.inner.cumulative_gas_used
+    }
```

Comment:
As we're dealing with a hierarchy of receipt types, this ensures a good user experience for the top-level receipt type. Otherwise you'd have to write `(**receipt).cumulative_gas_used`